### PR TITLE
Override native js functions window.alert and window.confirm 

### PR DESCRIPTION
**Pull request by orthagh**
(with identical behaviors)

For a prettier dialog : 

image

### CODE REVIEW

**Comment by tomolimo** [comment]
Is there a possible title for the dialog?

File: inc/html.class.php [hunk 1]
```diff
@@ -5554,6 +5556,105 @@ static function showCheckboxMatrix(array $columns, array $rows, array $options=a
...
+            // call asynchronous dialog
+            newConfirm(message);
+         }
+
+         // return early
+         return confirmed;
+      };");
+   }
 }
 ?>
```
Comment (newls) [inline inc/html.class.php]: why can't we like commits facebook-style ?

**Comment by tomolimo** [comment]
???

File: inc/html.class.php (hunk 1 above)
Comment (newls) [inline inc/html.class.php]: https://tracysite.files.wordpress.com/2013/09/like.jpg

Comment (tomolimo) [inline inc/html.class.php]: ah yes, I understand now :)

**Comment by orthagh** [comment]
Here is the optionnal caption :

```javascript
window.confirm('My confirm dialog', 'My title');

window.alert('My alert dialog', 'My title');
...

If not present, no title title is displayed for confirm. 
For alert, "Information" by default (title is always present)

**Comment by tomolimo** [comment]
Perfect!

**Comment by orthagh** [comment]
I'll keep this pr unmerged until Friday.

Waiting for suggestions and improvments

**Comment by tomolimo** [comment]
ok

File: inc/html.class.php (hunk 1 above)
Comment (newls) [inline inc/html.class.php]: haha @tomolimo i was scared of a tomahawklimo to be sent :p

Comment (tomolimo) [inline inc/html.class.php]: :)

**Comment by tomolimo** [comment]
There is a problem as confirm() is asynchronous and works with the last clicked element....
If there is no clicked element this doesn't work in this case confirm returns immediately.
Those dialog boxes MUST be synchronous by definition.

**Comment by tomolimo** [comment]
I don't have time to fix this now, but I can do it on Monday

**Comment by orthagh** [comment]
Yes. Native confirm function is synchronous and new implementation is asynchronous (no solution for that, i spent 3hours to tests various code).
The idea was to retrigger click on last element to resend the confirm dialog and the value returned.

It absolutely can't work if you don't have a user stimuli (generally 'click').

I let you think and test a solution but i doubt it will be possible.
I think we need to remove all confirm functions from code and replace by a generic function do display a jquery dialog but with an asynchronous mode (with a click listener on affected element).

**Comment by tomolimo** [comment]
I'll check that on Monday.
I've got already some ideas.
Regards
Tomolimo


-------- Original message --------
From: Alexandre Delaunay
Date:22/01/2016 16:54 (GMT+01:00)
To: glpi-project/glpi
Cc: "Moron, Olivier"
Subject: Re: [glpi] Override native js functions window.alert and window.confirm (#380)


Yes. Native confirm function is synchronous and new implementation is asynchronous (no solution for that, i spent 3hours to tests various code).
The idea was to retrigger click on last element to resend the confirm dialog and the value returned.

It absolutely can't work if you don't have a user stimuli (generally 'click').

I let you think and test a solution but i doubt it will be possible.
I think we need to remove all confirm functions from code and replace by a generic function do display a jquery dialog but with an asynchronous mode (with a click listener on affected element).

**Comment by tomolimo** [comment]
Questions: do we use Bootstrap?
and if we use it: which version of Bootstrap do we use in 0.91?

**Comment by orthagh** [comment]
No

**Comment by tomolimo** [comment]
At the end : can't changed asynch to synch :(

**Pull request merged**